### PR TITLE
Stress test outcome

### PR DESCRIPTION
**Pull request by aspacca**
Enhancement

## What does this PR do?

Address performance issues emerged during stress testing the lambda

## Why is it important?
We reduce that change of failure and missing ingestions with a proper scaled cluster

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.md`

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+### v0.12.0 - 2021/12/14
+##### Features
+* General performance refactoring as stress test outcome: [#48](https://github.com/elastic/elastic-serverless-forwarder/pull/48)
```
Comment (endorama) [inline CHANGELOG.md:3]:
```suggestion
* General performance refactoring after stress test outcome: [#48](https://github.com/elastic/elastic-serverless-forwarder/pull/48)
```

**Review by endorama — CHANGES_REQUESTED** [review]
Just a minor wording enhancement and a request: could you add to fafa0612c99330c0c1494913c3566f30af328a30 commit message some details about the change? I think they are very meaningful and merit to be added to the history for future reference.

**Comment by aspacca** [comment]
> could you add to fafa061 commit message some details about the change?

reworded as https://github.com/elastic/elastic-serverless-forwarder/pull/48/commits/a0dfb687eeb21fad2aae8752e8c63591f928e451

**Comment by aspacca** [comment]
@endorama fixed

**Comment by endorama** [comment]
I've just a question around lambda RAM limits. Does it mean that if I have a file larger than max Lambda memory I cannot ingest it directly into Elastic, but it requires to be split in smaller chunks? Just for clarity, with the current 10GB RAM limit I don't this as a concern

**Review by endorama — APPROVED** [review]

**Comment by aspacca** [comment]
> Does it mean that if I have a file larger than max Lambda memory I cannot ingest it directly into Elastic, but it requires to be split in smaller chunks?

that's correct, we could further improve and according to the length of the file given by the `head_object` request we can decide if using `get_object` with the eventual connection reset that we should find a way to handle, or `donwload_fileobj` in the current optimisation

**Pull request merged**